src/Parsers/DumpASTNode.h

```diff
@@ -86,6 +86,75 @@ inline void dumpAST(const IAST & ast, WriteBuffer & ostr, DumpASTNode * parent =
         dumpAST(*child, ostr, &dump);
 }
 
+class DumpASTNodeInDotFormat
+{
+public:
+    DumpASTNodeInDotFormat(const IAST & ast_, WriteBuffer * ostr_, bool root_ = true, const char * label_ = nullptr)
+        : ast(ast_), ostr(ostr_), root(root_), label(label_)
+    {
+        if (!ostr)
+            return;
+
+        if (root)
+            (*ostr) << "digraph " << (label ? String(label) : "") << "{\n    rankdir=\"UD\";\n";
```

Comment:
To support it in `play.html` we need to have eight just "digraph" on first line, like https://github.com/ClickHouse/ClickHouse/blob/master/src/QueryPipeline/printPipeline.h#L17

or update `play.html`, use condition `response.data[0][0].startsWith("digraph")`

https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/play.html#L579

Reply:
I think latter is better.